src/guidellm/dataset/synthetic.py

```diff
@@ -0,0 +1,243 @@
+import json
+import random
+from pathlib import Path
+from typing import Any, Dict, Iterable, Iterator, Optional, Tuple, Union
+
+import yaml
+from datasets import (
+    Dataset,
+    DatasetDict,
+    IterableDataset,
+    IterableDatasetDict,
+)
+from pydantic import BaseModel, Field
+from transformers import PreTrainedTokenizerBase
+
+from guidellm.dataset.creator import ColumnInputTypes, DatasetCreator
+from guidellm.utils import EndlessTextCreator, IntegerRangeSampler, check_load_processor
+
+__all__ = ["SyntheticDatasetCreator"]
+
+
+class SyntheticDatasetConfig(BaseModel):
+    prompt_tokens: int = Field(
+        description="The average number of text tokens generated for prompts.",
+        gt=0,
+    )
+    prompt_tokens_stdev: Optional[int] = Field(
+        description="The standard deviation of the tokens generated for prompts.",
+        gt=0,
+        default=None,
+    )
+    prompt_tokens_min: Optional[int] = Field(
+        description="The minimum number of text tokens generated for prompts.",
+        gt=0,
+        default=None,
+    )
+    prompt_tokens_max: Optional[int] = Field(
+        description="The maximum number of text tokens generated for prompts.",
+        gt=0,
+        default=None,
+    )
+    output_tokens: int = Field(
+        description="The average number of text tokens generated for outputs.",
+        gt=0,
+    )
+    output_tokens_stddev: Optional[int] = Field(
```

Comment:
```suggestion
    output_tokens_stdev: Optional[int] = Field(
```
nit: make consistent with prompt_tokens.